tests/test_server.py

```diff
@@ -56,23 +85,115 @@ def test_server_parse_channel_data(server, rtm_start_fixture):
 def test_server_parse_user_data(server, rtm_start_fixture):
     server.parse_user_data(rtm_start_fixture["users"])
     # Find user by Name
-    userbyname = server.users.find('fakeuser')
-    assert type(userbyname) == User
-    assert userbyname == "fakeuser"
-    assert userbyname != "someotheruser"
+    user_by_name = server.users.find('fakeuser')
+    assert type(user_by_name) == User
+    assert user_by_name == "fakeuser"
+    assert user_by_name != "someotheruser"
     # Find user by ID
-    userbyid = server.users.find('U10CX1234')
-    assert type(userbyid) == User
-    assert userbyid == "fakeuser"
-    assert userbyid.email == 'fakeuser@example.com'
+    user_by_id = server.users.find('U10CX1234')
+    assert type(user_by_id) == User
+    assert user_by_id == "fakeuser"
+    assert user_by_id.email == 'fakeuser@example.com'
     # Don't find invalid user
-    userbyid = server.users.find('invaliduser')
-    assert type(userbyid) != User
+    user_by_id = server.users.find('invaliduser')
+    assert user_by_id is None
 
 
 def test_server_cant_connect(server):
-    with pytest.raises(SlackLoginError):
-        reply = server.ping()
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            responses.POST,
+            "https://slack.com/api/rtm.start",
+            status=403,
+            json={"ok": False}
+        )
+
+        with pytest.raises(SlackConnectionError) as e:
+            server.rtm_connect()
+
+
+def test_reconnect_flag(server, rtm_start_fixture):
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            responses.POST,
+            "https://slack.com/api/rtm.start",
+            status=200,
+            json=rtm_start_fixture
+        )
+
+        server.rtm_connect(auto_reconnect=True)
```

Comment:
I think you want some tests that do `rsps.add` multiple times, so you can fail a few times and ensure you reconnect after N tries, right? (I might've missed this tho).

Essentially have ~3 `rsps.add`, the first two return a `4xx` and then you get a `200` for the third. That'll prove your reconnect handles errors properly 💪

Reply:
This reconnect logic only tests for when the socket is disconnected, then reconnected. It's not handling connection retries. That's part of another project 😄